server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -191,6 +200,18 @@ public void stop() {
         closed.set(true);
     }
 
+    /**
+     * Whether the poll() method of the normal Kafka connection should be called to avoid CommitFailedException considering
+     * the `max.poll.interval.ms` time, i.e., consumeTimeout value,
+     */
+    private boolean shouldNormPollBeCalled() {
+        if (this.lastTimeNormPollCalled != null) {
```

Comment:
I dislike the idea of keeping track of the time, as this does not guarantee a general solution to the problem. Should the Kafka config be changed to a smaller timeout, we will see the same problem. 

As a general solution to this, Kafka seems to support the concept of heartbeats... I would suggest to further explore the option to just sent a heartbeat for both normal and prio queue consumers at the beginning of each loop to prevent Kafka from rebalancing.

Reply:
Plus, as long as there are prio messages, we should not waste computational power on normal messages...

Reply:
I've found a better solution, which is using the `pause()` and `resume()` methods. See 54bde0c. I've tested with DummyPlugin and it works nicely!